src/ddbToEs/ddbToEsHelper.ts

```diff
@@ -66,16 +58,56 @@ export default class DdbToEsHelper {
         });
     }
 
-    async createIndexAndAliasIfNotExist(indexName: string, tenantId?: string) {
-        const alias = getAliasName(indexName, tenantId);
-        logger.debug('entering create index function');
+    // async createIndexAndAliasIfNotExist(resourceTypes: Set<string>) {
+    async createIndexAndAliasIfNotExist(aliases: { alias: string; index: string }[]) {
+        if (aliases.length === 0) {
+            return;
+        }
+
+        const uniqAliases = uniqWith(aliases, isEqual);
+        const listOfAliases = uniqAliases.map(x => x.alias);
+
+        const { body: allFound } = await this.ElasticSearch.indices.existsAlias({
+            name: listOfAliases,
+            expand_wildcards: 'all',
+        });
+        if (allFound) {
+            // All needed aliases exist
+            return;
+        }
+
+        logger.debug('There are missing aliases');
```

Comment:
To confirm we lazily create aliases? i.e. we have tenants: 1,2,3 

If a create Patient request comes in on tenant 1 the resulting ES cluster would have:
- index: `patient`
- alias: `patient-alias-tenant-1`

Reply:
that's correct